inductiva/stellarators/scenarios/coils/coils.py

```diff
@@ -237,3 +276,18 @@ def get_circular_curve_coefficients(toroidal_angle, major_radius, minor_radius):
     curve_coefficients[4, 1] = -minor_radius
 
     return curve_coefficients
+
+
+@StellaratorCoils.create_input_files.register
+def _(self, simulator: Simsopt, input_dir):  # pylint: disable=unused-argument
+    """Creates Simsopt simulation input files."""
+
+    coil_coefficients = [coil.curve_coefficients for coil in self.coils]
+    coil_currents = np.array([coil.current for coil in self.coils])
+
+    coil_coefficients_filename = os.path.join(input_dir,
+                                              COIL_COEFFICIENTS_FILENAME)
+    coil_currents_filename = os.path.join(input_dir, COIL_CURRENTS_FILENAME)
+
+    np.savez(coil_coefficients_filename, *coil_coefficients)
+    np.savez(coil_currents_filename, coil_currents)
```

Comment:
Why saving `*to_be_saved` in one case and not the other?

Reply:
Because `coil_coefficients` is a list of numpy arrays and `coil_currents` is just one numpy array. That was the way we decided to construct the classes initially.